cantor-phoenix/src/main/java/com/salesforce/cantor/phoenix/EventsOnPhoenix.java

```diff
@@ -0,0 +1,492 @@
+package com.salesforce.cantor.phoenix;
+
+import com.salesforce.cantor.Events;
+import com.salesforce.cantor.jdbc.AbstractBaseEventsOnJdbc;
+
+import javax.sql.DataSource;
+import java.io.IOException;
+import java.sql.*;
+import java.util.*;
+
+import org.apache.commons.collections4.map.MultiKeyMap;
+
+public class EventsOnPhoenix extends AbstractBaseEventsOnJdbc implements Events {
+
+    //TODO: in the future create tables and sequences in one batch
+    public EventsOnPhoenix(final DataSource dataSource) throws IOException {
+        super(dataSource);
+        String createMainTableSql = "create table if not exists cantor_events (timestampMillis BIGINT not null, " +
+                "id BIGINT not null, namespace VARCHAR, payload VARBINARY CONSTRAINT pk PRIMARY KEY (timestampMillis, id))";
+        String createMetadataTableSql = "create table if not exists cantor_events_m (timestampMillis BIGINT, " +
+                "id BIGINT, m_key VARCHAR, m_value VARCHAR, iid BIGINT PRIMARY KEY)";
+        String createDimensionsTableSql = "create table if not exists cantor_events_d (timestampMillis BIGINT, " +
+                "id BIGINT, d_key VARCHAR, d_value DOUBLE, iid BIGINT PRIMARY KEY)";
+        String createNamespaceTableSql = "create table if not exists cantor_events_namespace (namespace VARCHAR CONSTRAINT pk PRIMARY KEY (namespace))";
+        String createIDSeqSql = "create sequence if not exists cantor_events_id";
+        String createMIDSeqSql = "create sequence if not exists cantor_events_m_id";
+        String createDIDSeqSql = "create sequence if not exists cantor_events_d_id";
+        executeUpdate(createMainTableSql);
+        executeUpdate(createMetadataTableSql);
+        executeUpdate(createDimensionsTableSql);
+        executeUpdate(createNamespaceTableSql);
+        executeUpdate(createIDSeqSql);
+        executeUpdate(createMIDSeqSql);
+        executeUpdate(createDIDSeqSql);
+    }
+
+    public EventsOnPhoenix(final String path) throws IOException {
+        this(PhoenixDataSourceProvider.getDatasource(new PhoenixDataSourceProperties().setPath(path)));
+    }
+
+    //TODO: how to batchUpdate multiple statements w/ parameters
+    //TODO: if one event failed, are other events rolled back?
+    //TODO: if inserting into metadata/dimensions failed, how to roll back inserting into previous table(s)?
+    @Override
+    public void store(String namespace, Collection<Event> batch) throws IOException {
```

Comment:
Couple points:
- Take a look at other implementations of Events interface: all public methods are only responsible for validating arguments, and then they call into a private or protected method which actually does the job; prefixed with `do...`. Let's follow the same model here.
- Break up long functions into few smaller ones so it is more readable.
- Wherever you have repeatedly used literal strings (like `cantor_events_m`) let's create a constant for them.